worker/src/sql.rs

```diff
@@ -0,0 +1,216 @@
+
+use serde::de::DeserializeOwned;
+use serde_wasm_bindgen as swb;
+
+use crate::Error;
+use crate::Result;
+
+/// A value that can be stored in Durable Object SQL storage.
+///
+/// This enum represents the types that can be safely passed to SQL queries
+/// while maintaining type safety and proper conversion to JavaScript values.
+#[derive(Debug, Clone, PartialEq)]
+pub enum SqlStorageValue {
+    /// SQL NULL value
+    Null,
+    /// Boolean value
+    Boolean(bool),
+    /// 32-bit signed integer (safe for JavaScript Number conversion)
+    Integer(i32),
+    /// 64-bit floating point number
+    Float(f64),
+    /// UTF-8 string
+    String(String),
+    /// Binary data
+    Blob(Vec<u8>),
+}
+
+// From implementations for convenient conversion from Rust types
+impl From<bool> for SqlStorageValue {
+    fn from(value: bool) -> Self {
+        SqlStorageValue::Boolean(value)
+    }
+}
+
+impl From<i32> for SqlStorageValue {
+    fn from(value: i32) -> Self {
+        SqlStorageValue::Integer(value)
+    }
+}
+
+impl From<f64> for SqlStorageValue {
+    fn from(value: f64) -> Self {
+        SqlStorageValue::Float(value)
+    }
+}
+
+impl From<String> for SqlStorageValue {
+    fn from(value: String) -> Self {
+        SqlStorageValue::String(value)
+    }
+}
+
+impl From<&str> for SqlStorageValue {
+    fn from(value: &str) -> Self {
+        SqlStorageValue::String(value.to_string())
+    }
+}
+
+impl From<Vec<u8>> for SqlStorageValue {
+    fn from(value: Vec<u8>) -> Self {
+        SqlStorageValue::Blob(value)
+    }
+}
+
+impl<T> From<Option<T>> for SqlStorageValue
+where
+    T: Into<SqlStorageValue>,
+{
+    fn from(value: Option<T>) -> Self {
+        match value {
+            Some(v) => v.into(),
+            None => SqlStorageValue::Null,
+        }
+    }
+}
+
+// Convert SqlStorageValue to JsValue for passing to JavaScript
+impl From<SqlStorageValue> for JsValue {
+    fn from(val: SqlStorageValue) -> Self {
+        match val {
+            SqlStorageValue::Null => JsValue::NULL,
+            SqlStorageValue::Boolean(b) => JsValue::from(b),
+            SqlStorageValue::Integer(i) => JsValue::from(i),
+            SqlStorageValue::Float(f) => JsValue::from(f),
+            SqlStorageValue::String(s) => JsValue::from(s),
+            SqlStorageValue::Blob(bytes) => {
+                // Convert Vec<u8> to Uint8Array for JavaScript
+                let array = js_sys::Uint8Array::new_with_length(bytes.len() as u32);
+                array.copy_from(&bytes);
+                array.into()
+            }
+        }
+    }
+}
+
+/// Wrapper around the Workers `SqlStorage` interface exposed at `ctx.storage.sql`.
+///
+/// The API is intentionally minimal for now – additional helper utilities can be built on top
+/// as needed.
+#[derive(Clone, Debug)]
+pub struct SqlStorage {
+    inner: SqlStorageSys,
+}
+
+unsafe impl Send for SqlStorage {}
+unsafe impl Sync for SqlStorage {}
+
+impl SqlStorage {
```

Comment:
Seems this one is missing:

* prepare
* ingest
* setMaxPageCountForTest

Again, any reason to exclude these? It might just be worth investigating implementation / leaving a comment that these are pending implementation otherwise.

Reply:
got a lil spooked by https://github.com/cloudflare/workerd/pull/2722 https://github.com/cloudflare/workerd/blob/3d177d3/src/workerd/api/sql.h#L43-L52

left a TODO and reference to the sql.h. let me know how you'd like to proceed